### ISSUE TITLE
[BUG] Crash opening app: "An unhandled exception occured: 'N' is an invalid start of a value"

### ISSUE THREAD

**Issue by kwaegel** [RupertAvery/DiffusionToolkit#155]
**Describe the bug**
Opening the app causes a pop-up error with the message: "An unhandled exception occured: 'N' is an invalid start of a value. Path: $.safetensors-metadata | LineNumber: 65293 | BytePositionInLine: 28."

This may be a duplicate of [issue 116](https://github.com/RupertAvery/DiffusionToolkit/issues/116), but that one was marked as being related to images from Easy Diffusion, which I don't use.

**Version:**
1.4

**To Reproduce**
Open the app

**Expected behavior**
App should not crash.

**Source Image**
I'm assuming one of the recently generated images has invalid metadata that can't be parsed, or got added to the database in an invalid format, but the log does not contain enough information to figure out which one.

I have images generated using Automatic1111, ComfyUI, and InvokeAI.

**Source Image or Screenshots**
Screenshot and DiffusionToolkit.log attached.

<img width="311" alt="Screenshot 2023-12-10 175741" src="https://github.com/RupertAvery/DiffusionToolkit/assets/433893/5bda6f9e-1127-4c27-b72a-7f09a5b8eaad">

[DiffusionToolkit.log](https://github.com/RupertAvery/DiffusionToolkit/files/13630125/DiffusionToolkit.log)

**Comment by kwaegel** [comment]
I ran this under a debugger and tracked down the problem file. Turns out it's the Automatic1111 `cache.json` file that is failing to parse. Apparently one of the checkpoints contained invalid metadata, which got cached, and thus the cache file can't be parsed.

I heavily trimmed version of the JSON file shows the problem seems to be in parsing a literal `NaN` value that is not valid JSON. Trimming down the giant file to something readable gives roughly this nesting tree:
```
{
    "safetensors-metadata": {
        "checkpoint/XYZ.safetensors": {
            "value": {
                "prompt": {
                    "123": {
                        "is_changed": [
                            NaN
                        ]
                    }
```

This is probbaly a bug on the Automatic1111 side (writing invalid JSON data), but DiffusionToolkit should probably report which file is failing to parse, or maybe even automatically replace `NaN` with "null" if this is a common problem.
